controller/lib/core/sensors.cpp

```diff
@@ -151,29 +185,44 @@ void TVIntegrator::AddFlow(Time now, VolumetricFlow flow) {
 SensorReadings Sensors::GetSensorReadings() {
   // Flow rate is inhalation flow minus exhalation flow. Positive value is flow
   // into lungs, and negative is flow out of lungs.
+  auto now = Hal.now();
   auto patient_pressure = ReadPressureSensor(PATIENT_PRESSURE);
   auto inflow_delta = ReadPressureSensor(INFLOW_PRESSURE_DIFF);
   auto outflow_delta = ReadPressureSensor(OUTFLOW_PRESSURE_DIFF);
 
-  dp_inhale = inflow_delta.cmH2O();
-  dp_exhale = outflow_delta.cmH2O();
-  pressure = patient_pressure.cmH2O();
-
   VolumetricFlow inflow = PressureDeltaToFlow(inflow_delta);
   VolumetricFlow outflow = PressureDeltaToFlow(outflow_delta);
-  VolumetricFlow flow = inflow - outflow;
-  tv_integrator_.AddFlow(Hal.now(), flow);
+  VolumetricFlow uncorrected_flow = inflow - outflow;
+  VolumetricFlow corrected_flow = uncorrected_flow + flow_correction_;
+  uncorrected_tv_integrator_.AddFlow(now, uncorrected_flow);
+  tv_integrator_.AddFlow(now, corrected_flow);
 
+  // Set debug variables.
+  //
+  // TODO: This is repetitive and easy to mess up.  Can we improve the DebugVar
```

Comment:
DebugFloat should do it, no?

Reply:
With DebugFloat it would be the same code, just instead of `foo = bar` it's `foo.Set(bar)`, right?

Reply:
https://github.com/RespiraWorks/VentilatorSoftware/pull/463#discussion_r436441110